deeppavlov/dataset_readers/amazon_ecommerce_reader.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018 Neural Networks and Deep Learning lab, MIPT
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from pathlib import Path
+from typing import List, Any, Dict, Iterable, Tuple
+
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.data.utils import download_decompress, mark_done, is_done
+
+@register('amazon_ecommerce_reader')
+class AmazonEcommerceReader(DatasetReader):
+    """Class to download and load ecommerce data catalog"""
+
+    def read(self, data_path: str, catalog: str, **kwargs) -> Dict[str, List[Tuple[Any, Any]]]:
+        """Load data from specific catalog
+
+        Parameters:
+            data_path: path where to store the dataset
+            catalog: name of the specific part of the catalog
+        
+        Returns:
+            dataset: loaded dataset
+        """
+
+        ec_data_global: List[Any] = []
+        data_path = Path(expand_path(data_path))
+
+        if not is_done(data_path):
+            self._download_data(data_path)
+
+        if data_path.is_dir():
+            for fname in data_path.rglob("*.txt"):
+                if catalog in fname.name:
+                    ec_data_global = self._load_amazon_ecommerce_file(fname)
```

Comment:
If there are several files `in data_path.rglob("*.txt")`, only data from the last one will be used this way